schemachange/sc_callbacks.c

```diff
@@ -691,6 +690,8 @@ int scdone_callback(bdb_state_type *bdb_state, const char table[], void *arg,
         add_new_db = (db == NULL);
     }
 
+    assert(type != add || add_new_db == 1);
```

Comment:
I think we need to try the testsuite with optimization off because asserts don't trigger  at this time.

Reply:
okay will try